### ISSUE TITLE
Unable to submit build. Response code: 0

### ISSUE THREAD

**Issue by walker-ll** [fastlane/fastlane#9176]
# snapshot:

![image](https://cloud.githubusercontent.com/assets/7478867/26088413/b2e76ae0-3a29-11e7-9a92-59cae8c25dd8.png)

# log:
[10:42:53]: Uploading the build to Crashlytics Beta. Time for some ☕️.
[10:44:09]: 2017-05-16 10:42:53.326 submit Crashlytics: Crashlytics.framework/submit 1.3.5 (18)
2017-05-16 10:43:23.470 submit Crashlytics: Added Release Notes Successfully 200
2017-05-16 10:43:24.049 submit Crashlytics: Added Testers/Groups Successfully 200
2017-05-16 10:44:09.207 submit Crashlytics: Unable to submit build. Response code: 0
2017-05-16 10:44:09.608 submit Crashlytics: Failed to upload binary

+---------------------+--------------------------------------+
|                        Lane Context                        |
+---------------------+--------------------------------------+
| DEFAULT_PLATFORM    | ios                                  |
| PLATFORM_NAME       | ios                                  |
| LANE_NAME           | ios beta                             |
| CERT_FILE_PATH      | /Users/walker/work/iphonejallodesig  |
|                     | ner/iOS/G54B5QUK3Y.cer               |
| CERT_CERTIFICATE_ID | G54B5QUK3Y                           |
| SIGH_PROFILE_PATH   | /Users/walker/work/iphonejallodesig  |
|                     | ner/iOS/AdHoc_com.kujiale.JalloDesi  |
|                     | gner.mobileprovision                 |
| SIGH_PROFILE_PATHS  | ["/Users/walker/work/iphonejallodes  |
|                     | igner/iOS/AdHoc_com.kujiale.JalloDe  |
|                     | signer.mobileprovision"]             |
| SIGH_UDID           | 9e35211f-4859-4b4a-9c7b-d8fe7049a57  |
|                     | 1                                    |
| SIGH_UUID           | 9e35211f-4859-4b4a-9c7b-d8fe7049a57  |
|                     | 1                                    |
| SIGH_PROFILE_TYPE   | ad-hoc                               |
| IPA_OUTPUT_PATH     | /Users/walker/work/iphonejallodesig  |
|                     | ner/iOS/JalloDesigner.ipa            |
| DSYM_OUTPUT_PATH    | /Users/walker/work/iphonejallodesig  |
|                     | ner/iOS/JalloDesigner.app.dSYM.zip   |
| XCODEBUILD_ARCHIVE  | /Users/walker/Library/Developer/Xco  |
|                     | de/Archives/2017-05-16/JalloDesigne  |
|                     | r 2017-05-16 10.16.33.xcarchive      |
+---------------------+--------------------------------------+
[10:49:13]: No route to host - connect(2) for "clouderrorreporting.googleapis.com" port 443

+------+---------------------+-------------+
|             fastlane summary             |
+------+---------------------+-------------+
| Step | Action              | Time (in s) |
+------+---------------------+-------------+
| 1    | Verifying required  | 0           |
|      | fastlane version    |             |
| 2    | default_platform    | 0           |
| 3    | cert                | 5           |
| 4    | sigh                | 7           |
| 5    | gym                 | 1583        |
| 6    | ../commit.sh        | 0           |
| 💥   | crashlytics         | 76          |
+------+---------------------+-------------+

[10:49:13]: fastlane finished with errors

# fastlane env:

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                           |
| --------------------------- | ----------------------------------------------- |
| OS                          | 10.12                                           |
| Ruby                        | 2.2.4                                           |
| Bundler?                    | false                                           |
| Git                         | git version 2.10.1 (Apple Git-78)               |
| Installation Source         | /usr/local/lib/fastlane_lib/bundle/bin/fastlane |
| Host                        | Mac OS X 10.12 (16A323)                         |
| Ruby Lib Dir                | /usr/local/lib/fastlane_lib/bundle/lib          |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                      |
| Is contained                | false                                           |
| Is homebrew                 | false                                           |
| Is installed via Fabric.app | true                                            |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/     |
| Xcode Version               | 8.2.1                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.99.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    #cocoapods
    ENV["CRASHLYTICS_NOTES_PATH"] = "node.txt"
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  # desc "Submit a new Beta Build to Apple TestFlight"
  # desc "This will also make sure the profile is up to date"
  # lane :beta do
  #   # match(type: "appstore") # more information: https://codesigning.guide
  #   gym(scheme: "JalloDesigner") # Build your app - more options available
  #   pilot

  #   # sh "your_script.sh"
  #   # You can also use other beta testing services here (run `fastlane actions`)
  # end
  desc "Submit a new Beta Build to Fabric"
  lane :beta do
    cert
    sigh(adhoc: true)
    gym(
      workspace: "JalloDesigner.xcworkspace",
      scheme: "JalloDesigner-beta",
      configuration: "ForBeta",
      clean: false
    )
    sh "../commit.sh"
    # sh "../build_react_native.sh"
    crashlytics(
      api_token: "eabdaed789c98cba8d263da9cda89aab40083abe",
      build_secret: "97efa54ab57b4da53e8855a88e3d75615934131bfde995bcf7605a7f79fda16f",
      groups: "qa-team",
      # notes: "Automatic iOS Build"
      # notes_path: "../note.txt"
    )
    sh "../bumpBuilderNumber.sh"
  end

  lane :fabric do
    sh "../commit.sh"
    crashlytics(
      api_token: "eabdaed789c98cba8d263da9cda89aab40083abe",
      build_secret: "97efa54ab57b4da53e8855a88e3d75615934131bfde995bcf7605a7f79fda16f",
      groups: "qa-team",
      # notes: "Automatic iOS Build"
      # notes_path: note.txt
    )
    sh "../bumpBuilderNumber.sh"
  end

  lane :rn do
    # sh "../commit.sh"
    sh "../build_react_native.sh"
  end

  desc "Deploy a new version to the App Store"
  lane :app do
    # match(type: "appstore")
    # snapshot
    gym(scheme: "JalloDesigner") # Build your app - more options available
    deliver(force: true)
    # frameit
  end

  lane :submit do
    # match(type: "appstore")
    # snapshot
    deliver(force: true)
    # frameit
  end

  desc "Deploy a new version to the App Store"
  lane :appstoreAdhoc do
    # match(type: "appstore")
    # snapshot
    gym(scheme: "JalloDesigner") # Build your app - more options available
    # deliver(force: true)ß
    # frameit
  end
  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "com.kujiale.JalloDesigner" # The bundle identifier of your app
apple_id "forcemap@163.com" # Your Apple email address

team_id "BZM86V26D4"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.30.2  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.1        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.4        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.7        |
| terminal-notifier         | 1.7.2        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.3.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.28       |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
</details>

*generated on:* **2017-05-16**
</details>

### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines]

(https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

[INSERT OUTPUT HERE]

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

[INSERT OUTPUT HERE]

**Comment by walker-ll** [comment]
i have updated the latest fastlane 2.30.2

**Comment by mfurtak** [comment]
This extra bit of logging is showing that the crash report also failed to be submitted, but to a different service/domain.

`[10:49:13]: No route to host - connect(2) for "clouderrorreporting.googleapis.com" port 443`

This makes me think that network connectivity is not good from that machine. Can you make sure that `curl https://apps.crashlytics.com` can succeed from that machine?

**Comment by damoncoo** [comment]
@mfurtak  I got the same issue sometimes when I run fastlane on jenkins.
The outout of Jenkins is as belows:

```
[32m# fastlane 2.35.1 is available. You are on 2.33.0.[0m
[32m# You should use the latest version.[0m
[32m# Please update using `fastlane update_fastlane`.[0m
[32m#######################################################################[0m

[32m2.35.1 Scan hotfix[0m
* [scan] Use pre-2.34.0 strategy if temp junit output is missing (#9276) via Keaton Greve
* Fix puts crash on linux (#9279) via David Ohayon
* Warn user when before_all, after_all or error block overwrite each other (#9284) via Felix Krause
* Add helpful output when running `pilot` without specifying an action (#9283) via Felix Krause
* Update links to new docs page in crashlytics beta class (#9264) via Felix Krause

[32m2.35.0 Update 2-factor to use new olympus API[0m
* Update spaceship to support 2 factor on the new `olympus` API endpoint (#9266) via Felix Krause and Joshua Liebowitz
* Remove duplicate auth code in spaceship (#9253) via Felix Krause
* Add `:configuration` option for slather action (#8825) via Takashi Hasegawa

[32m2.34.0 Introducing nightly builds[0m
* Add opt-in for nightly builds using `update_fastlane(nightly: true)` (#8720) via Felix Krause
* Catch crashes in the crash reporter after being unreachable in China (#9263) via David Ohayon
* [scan] Reduce run time by removing report generation command as separate process (#7365) via Keaton Greve
* Create log archive directory in case it doesn’t exist (#9246) via Lyndsey Ferguson
* [slack] Fix crash when defining FL_SLACK_DEFAULT_PAYLOADS (#9217) via Liam Nichols
* Skip error message when no simulator wasn't provided (#9240) via Felix Krause
* Add a EOF newline in the pluginfile content if its not already there (#8942) via Josh Lesch
* Rescue exceptions when crash reporter cannot write out status file (#9245) via Mark Pirri
* TestFairy option checking only checked for symbols, not strings (#9257) via Joshua Liebowitz

---

Please help us test early releases of fastlane by opting into nightly builds 🌃

```ruby
before_all do
  update_fastlane(nightly: true)
end
```

![image](https://cloud.githubusercontent.com/assets/869950/26415847/baa85d76-4068-11e7-8251-5b0f3ded58b3.png)

[32mTo see all new releases, open https://github.com/fastlane/fastlane/releases[0m

[32mPlease update using `fastlane update_fastlane`[0m
fastlane could not check for updates error: execution expired
[18:31:47]: [32m-------------------------------------------------[0m
[18:31:47]: [32m--- Step: Verifying required fastlane version ---[0m
[18:31:47]: [32m-------------------------------------------------[0m
[18:31:47]: Your fastlane version 2.33.0 matches the minimum requirement of 2.17.0  ✅
[18:31:47]: [32m------------------------------[0m
[18:31:47]: [32m--- Step: default_platform ---[0m
[18:31:47]: [32m------------------------------[0m
[18:31:47]: [32mDriving the lane 'ios beta' 🚀[0m
[18:31:47]: [32m------------------------------[0m
[18:31:47]: [32m--- Step: register_devices ---[0m
[18:31:47]: [32m------------------------------[0m
[18:31:53]: Fetching list of currently registered devices...
+------------------+----------+
|        [33mLane Context[0m         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios beta |
+------------------+----------+
[18:37:26]: [31mNo route to host - connect(2) for "clouderrorreporting.googleapis.com" port 443[0m

+------+-------------------------------------+-------------+
|                     [32mfastlane summary[0m                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 💥   | [31mregister_devices[0m                    | 8           |
+------+-------------------------------------+-------------+

[18:37:27]: [31mfastlane finished with errors[0m
[32m#######################################################################[0m
[32m# fastlane 2.35.1 is available. You are on 2.33.0.[0m
[32m# You should use the latest version.[0m
[32m# Please update using `fastlane update_fastlane`.[0m
[32m#######################################################################[0m
[32m2.35.1 Scan hotfix[0m
* [scan] Use pre-2.34.0 strategy if temp junit output is missing (#9276) via Keaton Greve
* Fix puts crash on linux (#9279) via David Ohayon
* Warn user when before_all, after_all or error block overwrite each other (#9284) via Felix Krause
* Add helpful output when running `pilot` without specifying an action (#9283) via Felix Krause
* Update links to new docs page in crashlytics beta class (#9264) via Felix Krause

[32m2.35.0 Update 2-factor to use new olympus API[0m
* Update spaceship to support 2 factor on the new `olympus` API endpoint (#9266) via Felix Krause and Joshua Liebowitz
* Remove duplicate auth code in spaceship (#9253) via Felix Krause
* Add `:configuration` option for slather action (#8825) via Takashi Hasegawa

[32m2.34.0 Introducing nightly builds[0m
* Add opt-in for nightly builds using `update_fastlane(nightly: true)` (#8720) via Felix Krause
* Catch crashes in the crash reporter after being unreachable in China (#9263) via David Ohayon
* [scan] Reduce run time by removing report generation command as separate process (#7365) via Keaton Greve
* Create log archive directory in case it doesn’t exist (#9246) via Lyndsey Ferguson
* [slack] Fix crash when defining FL_SLACK_DEFAULT_PAYLOADS (#9217) via Liam Nichols
* Skip error message when no simulator wasn't provided (#9240) via Felix Krause
* Add a EOF newline in the pluginfile content if its not already there (#8942) via Josh Lesch
* Rescue exceptions when crash reporter cannot write out status file (#9245) via Mark Pirri
* TestFairy option checking only checked for symbols, not strings (#9257) via Joshua Liebowitz

---

Please help us test early releases of fastlane by opting into nightly builds 🌃

```ruby
before_all do
  update_fastlane(nightly: true)
end
```

![image](https://cloud.githubusercontent.com/assets/869950/26415847/baa85d76-4068-11e7-8251-5b0f3ded58b3.png)

[32mTo see all new releases, open https://github.com/fastlane/fastlane/releases[0m

[32mPlease update using `fastlane update_fastlane`[0m
/Users/xiaowu/.fastlane/bin/bundle/lib/ruby/2.2.0/net/http.rb:879:in `initialize': No route to host - connect(2) for "clouderrorreporting.googleapis.com" port 443 (Faraday::ConnectionFailed)
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/2.2.0/net/http.rb:879:in `open'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/2.2.0/net/http.rb:879:in `block in connect'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/2.2.0/timeout.rb:73:in `timeout'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/2.2.0/net/http.rb:878:in `connect'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/2.2.0/net/http.rb:863:in `do_start'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/2.2.0/net/http.rb:852:in `start'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/2.2.0/net/http.rb:1375:in `request'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/faraday-0.11.0/lib/faraday/adapter/net_http.rb:80:in `perform_request'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/faraday-0.11.0/lib/faraday/adapter/net_http.rb:38:in `block in call'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/faraday-0.11.0/lib/faraday/adapter/net_http.rb:85:in `with_net_http_connection'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/faraday-0.11.0/lib/faraday/adapter/net_http.rb:33:in `call'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/faraday-0.11.0/lib/faraday/request/url_encoded.rb:15:in `call'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/faraday-0.11.0/lib/faraday/rack_builder.rb:139:in `build_response'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/faraday-0.11.0/lib/faraday/connection.rb:377:in `run_request'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/faraday-0.11.0/lib/faraday/connection.rb:177:in `post'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.33.0/fastlane_core/lib/fastlane_core/crash_reporter/crash_reporter.rb:77:in `send_report'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.33.0/fastlane_core/lib/fastlane_core/crash_reporter/crash_reporter.rb:29:in `report_crash'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.33.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:88:in `rescue in run!'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.33.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:97:in `run!'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.4/lib/commander/delegates.rb:15:in `run!'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.33.0/fastlane/lib/fastlane/commands_generator.rb:303:in `run'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.33.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.33.0/fastlane/lib/fastlane/cli_tools_distributor.rb:66:in `take_off'
	from /Users/xiaowu/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.33.0/bin/fastlane:20:in `<top (required)>'
	from /Users/xiaowu/.fastlane/bin/bundle/bin/fastlane:22:in `load'
	from /Users/xiaowu/.fastlane/bin/bundle/bin/fastlane:22:in `<main>'
```

curl https://apps.crashlytics.com ,it can get content

**Comment by taquitos** [comment]
@damoncoo please update to the newest _fastlane_ release on that machine. We updated recently to fix the issue of not being able to connect to `clouderrorreporting.googleapis.com` That should fix your problem 👍
